internal/worker_server/server.go

```diff
@@ -0,0 +1,58 @@
+package workerserver
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/flightctl/flightctl/internal/config"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/tasks"
+	"github.com/flightctl/flightctl/pkg/queues"
+	"github.com/sirupsen/logrus"
+)
+
+type Server struct {
+	cfg   *config.Config
+	log   logrus.FieldLogger
+	store store.Store
+}
+
+// New returns a new instance of a flightctl server.
+func New(
+	cfg *config.Config,
+	log logrus.FieldLogger,
+	store store.Store,
+) *Server {
+	return &Server{
+		cfg:   cfg,
+		log:   log,
+		store: store,
+	}
+}
+
+func (s *Server) Run() error {
+	s.log.Println("Initializing async jobs")
+	provider := queues.NewAmqpProvider(s.cfg.Queue.AmqpURL, s.log)
+	publisher, err := tasks.FleetQueuePublisher(provider)
```

Comment:
So we have a single queue so far, although in that queue we send some other tasks that aren't only fleet related.

Should it be TaskQueue or TaskQueuePublisher instead?

otherwise it is confusing.

Reply:
I will rename